### PR TITLE
[ISSUE-2673] Improve tests in PhysicalPlanSerializeTest

### PR DESCRIPTION
**Pull request by weizihan0110**
# 1.Extract common codes into methods like testTwoSerializeMethodAndDeserialize and serializePlan.
image

# 2.Alter tests to test "void serialize(DataOutputStream stream)" method in some plans.
image
image

### CODE REVIEW

**Review by qiaojialin — APPROVED** [review]

**Review by jixuan1989 — CHANGES_REQUESTED** [review]

File: Jenkinsfile [hunk 1]
```diff
@@ -106,7 +106,7 @@ pipeline {
                     // Then run the analysis
                     // 'my-sonarcloud-token' needs to be defined for this job and contains the user token
                     withCredentials([string(credentialsId: 'xiangdong-iotdb-sonarcloud-token', variable: 'SONAR_TOKEN')]) {
-                        sh "mvn verify sonar:sonar -Dsonar.branch.name=master -Dsonar.host.url=https://sonarcloud.io -Dsonar.organization=apache -Dsonar.projectKey=apache_incubator-iotdb -Dsonar.login=${SONAR_TOKEN} -DskipTests"
+                        sh "mvn -X verify sonar:sonar -Dsonar.branch.name=master -Dsonar.host.url=https://sonarcloud.io -Dsonar.organization=apache -Dsonar.projectKey=apache_incubator-iotdb -Dsonar.login=${SONAR_TOKEN} -DskipTests"
```
[the remark this answers is not in the document]
Comment (weizihan0110) [reply Jenkinsfile:109]: > do not add "-X"

I wanna see what causes the error. Just for the detailed debug info. I'll delete it after solving the problem.

**Comment by GLBB** [comment]
I also have this problem
```
Main Linux / ubuntu (11) (pull_request) Failing after 39m — ubuntu (11)
```
if solved, please help me

**Comment by jixuan1989** [comment]
I suspect it is caused by some permission issue. 
As we are not using SonarCloud token directly. Instead, we are using Github secret Token. 
You can omit it before I find a solution (have to ask help from Apache Infra).

**Comment by weizihan0110** [comment]
> I suspect it is caused by some permission issue.
> As we are not using SonarCloud token directly. Instead, we are using Github secret Token.
> You can omit it before I find a solution (have to ask help from Apache Infra).

Hi, @jixuan1989. I've already removed "-X" param. And just SonarCloud Report Error left.

**Review by HTHou — APPROVED** [review]

**Pull request merged**